rust/datafusion/src/execution/context.rs

```diff
@@ -73,26 +74,42 @@ impl ExecutionContext {
         }
     }
 
-    pub fn register_datasource(&mut self, name: &str, ds: Rc<RefCell<DataSource>>) {
-        self.datasources.borrow_mut().insert(name.to_string(), ds);
+    pub fn register_csv(
```

Comment:
We should (maybe at some point later) document to these public methods.

Reply:
Yes. I'm actually working on a refactor of data source providers and I want users to be able to register their own custom data source providers (so they are not dependent on our release cycle to add new ones) and will add documentation as part of that PR.

Reply:
Sounds good.

Reply:
I just added docs to all methods in ExecutionContext